### PR TITLE
fix: support fetching manifest in oci image layout

### LINKED ISSUE #764: `oras manifest fetch` fails on oci image layout target
### What happened in your environment?

I have an image tagged `test` stored in `oci-root` folder, I am able to pull it but unable to use `oras manifest fetch` to get the manifest content.

### What did you expect to happen?

`oras manifest fetch` should show the manifest content of `oci-root\:test`

### How can we reproduce it?

```
$ echo image > image

$ oras push oci-root/:test image --oci-layout

$ oras repo tags oci-root --oci-layout
test

$ oras pull oci-root/:test --oci-layout
Downloading 0d566894635a image
Downloaded  0d566894635a image
Pulled [oci-layout] oci-root/:test
Digest: sha256:b311ca6afe69cec37e3cf3d6a56369be78de91bb0523dbdf4c40a23a65d7a0d1       

$ oras manifest fetch oci-root/:test --oci-layout
Error: not found
```

### What is the version of your ORAS CLI?

I build it from the main branch of ORAS (last commit: 735ce355ef351ee03cb044e5ef03f485c05d1953)

### What is your OS environment?

20.04

### Are you willing to submit PRs to fix it?

- [X] Yes, I am willing to fix it.

### PR DESCRIPTION
**Pull request by qweeah**
Resolves #764

Signed-off-by: Billy Zha <jinzha1@microsoft.com>

### CODE REVIEW

File: cmd/oras/manifest/fetch.go [hunk 1]
```diff
@@ -113,18 +114,18 @@ func fetchManifest(opts fetchOptions) (fetchErr error) {
 		// fetch manifest descriptor only
 		fetchOpts := oras.DefaultResolveOptions
 		fetchOpts.TargetPlatform = opts.Platform.Platform
-		desc, err = oras.Resolve(ctx, src, opts.RawReference, fetchOpts)
+		desc, err = oras.Resolve(ctx, src, opts.Reference, fetchOpts)
```
[the remark this answers is not in the document]
Comment (qweeah) [reply cmd/oras/manifest/fetch.go:117]: Indeed, E2E tests should be added.

OCI image layout support is a new feature and tests are not there yet. Let me open an issue for tracking.

[the remark this answers is not in the document]
Comment (qweeah) [reply cmd/oras/manifest/fetch.go:117]: Added an issue https://github.com/oras-project/oras/issues/767, it should be done before 1.0.0 release.

File: cmd/oras/manifest/fetch.go [hunk 2]
```diff
@@ -113,18 +114,18 @@ func fetchManifest(opts fetchOptions) (fetchErr error) {
...
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to find %q in %q: %w", opts.Reference, opts.RawReference, err)
```
Comment (shizhMSFT) [inline cmd/oras/manifest/fetch.go:119]: What will this error message look like? It seems that it contains duplicated messages.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**